### PR TITLE
[WIP] Move settings.py to habitat_sim.utils

### PR DESCRIPTION
**Pull request by jaraujo98**
## Motivation and Context

One task I have to do frequently is to start a “default” simulator (mostly for writing custom viewers). The only option I am aware of is to use the code in `habitat-sim/examples/settings.py` to get a default configuration. Having to rely on this file has been a pain point. I have to either run my scripts from the examples folder or the sim folder (this leads to inconsistent imports: on some files the statement is `from settings import ...`, on others it is `from examples.settings import ...`). I can also play around with `sys.path` or `PYTHONPATH`, but this not a viable solution since it can clash with other packages that also have folders named `examples` or files named `settings`.

This PR moves the `settings.py` file into `habitat_sim.utils`, thus making its functionality unambiguously available everywhere. I kept a file named `settings.py` inside `examples` that simply imports the original settings file to make sure existing scripts that rely on it don't break.

## How Has This Been Tested

Tests (some of which rely on this file) pass. The existing examples also continue working.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src_python/habitat_sim/utils/settings.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    "ortho_depth_sensor": False,  # Orthographic depth sensor (default: OFF)
+    "ortho_semantic_sensor": False,  # Orthographic semantic sensor (default: OFF)
+    "fisheye_rgba_sensor": False,
+    "fisheye_depth_sensor": False,
+    "fisheye_semantic_sensor": False,
+    "equirect_rgba_sensor": False,
+    "equirect_depth_sensor": False,
+    "equirect_semantic_sensor": False,
+    "seed": 1,
+    "silent": False,  # do not print log info (default: OFF)
```
Comment (aclegg3) [inline src_python/habitat_sim/utils/settings.py:28]: This one is more relevant for examples as logging is configurable externally.

File: src_python/habitat_sim/utils/settings.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import habitat_sim
+import habitat_sim.agent
+
+default_sim_settings = {
+    "max_frames": 1000,
```
Comment (aclegg3) [inline src_python/habitat_sim/utils/settings.py:9]: This is an example setting

File: src_python/habitat_sim/utils/settings.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    sim_cfg = habitat_sim.SimulatorConfiguration()
+    if "scene_dataset_config_file" in settings:
+        sim_cfg.scene_dataset_config_file = settings["scene_dataset_config_file"]
+    sim_cfg.frustum_culling = settings.get("frustum_culling", False)
+    if "enable_physics" in settings:
+        sim_cfg.enable_physics = settings["enable_physics"]
+    if "physics_config_file" in settings:
+        sim_cfg.physics_config_file = settings["physics_config_file"]
+    if not settings["silent"]:
+        print("sim_cfg.physics_config_file = " + sim_cfg.physics_config_file)
```
Comment (aclegg3) [inline src_python/habitat_sim/utils/settings.py:42]: This should be moved to example make_cfg()

File: src_python/habitat_sim/utils/settings.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    "fisheye_semantic_sensor": False,
+    "equirect_rgba_sensor": False,
+    "equirect_depth_sensor": False,
+    "equirect_semantic_sensor": False,
+    "seed": 1,
+    "silent": False,  # do not print log info (default: OFF)
+}
+
+# build SimulatorConfiguration
+def make_cfg(settings):
```
Comment (aclegg3) [inline src_python/habitat_sim/utils/settings.py:32]: We should have an example specific make_cfg() with some of this functionality.

File: src_python/habitat_sim/utils/settings.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+        ),
+    }
+
+    # override action space to no-op to test physics
+    if sim_cfg.enable_physics:
+        agent_cfg.action_space = {
+            "move_forward": habitat_sim.agent.ActionSpec(
+                "move_forward", habitat_sim.agent.ActuationSpec(amount=0.0)
+            )
+        }
```
Comment (aclegg3) [inline src_python/habitat_sim/utils/settings.py:214]: This override is an example setting.

File: examples/demo_runner.py [hunk 6]
```diff
@@ -12,14 +12,14 @@
...
 from PIL import Image
-from settings import default_sim_settings, make_cfg
 
 import habitat_sim
 import habitat_sim.agent
 import habitat_sim.utils.datasets_download as data_downloader
 from habitat_sim.nav import ShortestPath
 from habitat_sim.physics import MotionType
 from habitat_sim.utils.common import d3_40_colors_rgb, quat_from_angle_axis
+from habitat_sim.utils.settings import default_sim_settings, make_cfg
```
Comment (aclegg3) [inline examples/demo_runner.py:22]: If both default_sim_settings and make_cfg are extended in examples/settings, this should probably be reverted.

**Review by aclegg3 — COMMENTED** [review]
Great to see the tests no longer pulling in examples. Mostly I suggest we extend both default_settings and make_cfg() in examples and move some things there.

File: src_python/habitat_sim/utils/settings.py (hunk 3 above)
Comment (jaraujo98) [reply src_python/habitat_sim/utils/settings.py:42]: Only the if statement concerning `settings["silent"]`, right? Just checking.

Comment (aclegg3) [reply src_python/habitat_sim/utils/settings.py:42]: Yes :+1:

**Review by aclegg3 — APPROVED** [review]
LGTM

**Pull request merged**